src/document-click.js

```diff
@@ -68,40 +72,68 @@ export class ClickHandler {
     /** @private @const {!./service/history-impl.History} */
     this.history_ = historyFor(this.win);
 
+    /** @private @const {!./service/url-replacements-impl.UrlReplacements} */
+    this.urlReplacements_ = urlReplacementsFor(this.win);
+
     const platform = platformFor(this.win);
     /** @private @const {boolean} */
     this.isIosSafari_ = platform.isIos() && platform.isSafari();
 
+    /** @private {!Array<!function(!Event)|undefined>} */
+    this.boundHandlers_ = [];
+
     // Only intercept clicks when iframed.
     if (this.viewer_.isIframed() && this.viewer_.isOvertakeHistory()) {
-      /** @private @const {!function(!Event)|undefined} */
-      this.boundHandle_ = this.handle_.bind(this);
+      this.boundHandlers_.push(this.handle_.bind(this, false));
       this.win.document.documentElement.addEventListener(
-          'click', this.boundHandle_);
+          'click', this.boundHandlers_[this.boundHandlers_.length - 1]);
     }
+    // Add capture phase click handler for anchor target href expansion.
+    this.boundHandlers_.push(this.handle_.bind(this, true));
+    this.win.document.documentElement.addEventListener(
+        'click', this.boundHandlers_[this.boundHandlers_.length - 1], true);
   }
 
   /**
    * Removes all event listeners.
    */
   cleanup() {
-    if (this.boundHandle_) {
+    this.boundHandlers_.forEach(handler => {
       this.win.document.documentElement.removeEventListener(
-          'click', this.boundHandle_);
-    }
+          'click', handler);
+    });
+    this.boundHandlers_ = [];
   }
 
   /**
-   * Intercept any click on the current document and prevent any
-   * linking to an identifier from pushing into the history stack.
+   * Click event handler which on bubble propagation intercepts any click on the
+   * current document and prevent any linking to an identifier from pushing into
+   * the history stack; on capture propagation expands anchor href.
+   * @param {boolean} isCapture
    * @param {!Event} e
    */
-  handle_(e) {
+  handle_(isCapture, e) {
     onDocumentElementClick_(e, this.viewport_, this.history_,
-        this.isIosSafari_);
+        this.urlReplacements_, this.isIosSafari_, isCapture);
   }
 }
 
+/**
+ * Locate first element with given tag name within event path from shadowRoot.
+ * @param {!Event} e
+ * @param {!string} tagName
+ * @return {?Element}
+ * @visibleForTesting
+ */
+export function getElementByTagNameFromEventShadowDomPath_(e, tagName) {
```

Comment:
No because of how shadow DOM does event target rewriting (see http://www.html5rocks.com/en/tutorials/webcomponents/shadowdom-301/#toc-events).  Event.target will always be the host element while we want to determine the element within the shadom DOM in which case event.path should be used.